File: doacao_app_java/src/main/java/schemas/ProcessoAdocao.java
package schemas;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

public class ProcessoAdocao {
    private Integer idPAdocao;
    private Animal animal;
    private Adotante adotante;
    private StatusProcesso status;
    private LocalDate dataIncio;
    private List<EtapaProcesso> etapas = new ArrayList<>();
    private List<Mensagem> mensagems = new ArrayList<>();
    private Integer id_animal;
    private Integer id_adotante;

    public ProcessoAdocao(Integer idPAdocao, Animal animal, Adotante adotante, StatusProcesso status, LocalDate dataIncio, Integer id_animal, Integer id_adotante) {
        this.idPAdocao = idPAdocao;
        this.animal = animal;
        this.adotante = adotante;
        this.status = status;
        this.dataIncio = dataIncio;
        this.id_animal = id_animal;
        this.id_adotante = id_adotante;
    }

    public ProcessoAdocao() {

    }

    public Integer getIdPAdocao() {
        return idPAdocao;
    }

    public void setIdPAdocao(Integer idPAdocao) {
        this.idPAdocao = idPAdocao;
    }

    public Animal getAnimal() {
        return animal;
    }

    public void setAnimal(Animal animal) {
        this.animal = animal;
    }

    public Adotante getAdotante() {
        return adotante;
    }

    public void setAdotante(Adotante adotante) {
        this.adotante = adotante;
    }

    public StatusProcesso getStatus() {
        return status;
    }

    public void setStatus(StatusProcesso status) {
        this.status = status;
    }

    public LocalDate getDataIncio() {
        return dataIncio;
    }

    public void setDataIncio(LocalDate dataIncio) {
        this.dataIncio = dataIncio;
    }

    public List<EtapaProcesso> getEtapas() {
        return etapas;
    }

    public void setEtapas(List<EtapaProcesso> etapas) {
        this.etapas = etapas;
    }

    public List<Mensagem> getMensagems() {
        return mensagems;
    }

    public void setMensagems(List<Mensagem> mensagems) {
        this.mensagems = mensagems;
    }

    public void adicionarEtapa(EtapaProcesso etapa) {
        this.etapas.add(etapa);
    }

    public void mudarStatus(StatusProcesso novoStatus) {
        this.status = novoStatus;
        System.out.println("Procecsso" + idPAdocao + "agora está em" + novoStatus);
    }

    public Integer getId_animal() {
        return id_animal;
    }

    public void setId_animal(Integer id_animal) {
        this.id_animal = id_animal;
    }

    public Integer getId_adotante() {
        return id_adotante;
    }

    public void setId_adotante(Integer id_adotante) {
        this.id_adotante = id_adotante;
    }
}

File: doacao_animal_python/schemas/processo_adocao.py
from datetime import datetime as dt
from typing import List

from .adotante import Adotante
from .animal import Animal
from .etapa_processo import EtapaProcesso
from .mensagem import Mensagem
from .status_processo import StatusProcesso

class ProcessoAdocao:
    def __init__(self, idPAdocao: int, animal: Animal, adotante: Adotante, statusProcesso: StatusProcesso, dataInicio: dt, id_animal: int, id_adotante: int, etapas: List[EtapaProcesso] = [], mensagens: List[Mensagem] = []):
        self.idPAdocao = idPAdocao
        self.animal = animal
        self.adotante = adotante
        self.statusProcesso = statusProcesso
        self.dataInicio = dataInicio
        self.etapas = etapas
        self.mensagens = mensagens
        self.id_animal = id_animal
        self.id_adotante = id_adotante
        
    def adicionarEtapa(self, etapa: EtapaProcesso):
        self.etapas.append(etapa)
    
    def mudarStatus(self, novo_status: StatusProcesso):
        self.statusProcesso = novo_status
        print(f"Status do processo {self.idPAdocao} atualizado para: {self.statusProcesso.value}")
    